lib/tic_tac_toe/web/router.ex

```diff
@@ -1,62 +1,54 @@
 defmodule TicTacToe.Web.Router do
   use Plug.Router
-  import TicTacToe.Web.GameSessionPlug
-  alias TicTacToe.Web.GameSessionPlug
+  alias TicTacToe.Web.GameStateStringifier
   alias TicTacToe.Web.View
 
-  @secret String.duplicate("abcdef0123456789", 8)
-
-  plug Plug.Session, store: :cookie, key: "_my_app_session",
-                     encryption_salt: "cookie store encryption salt",
-                     signing_salt: "cookie store signing salt",
-                     key_length: 64, log: :debug
-
   plug Plug.Parsers, parsers: [:urlencoded, :multipart]
   plug Plug.Static, at: "/public", from: :elixir_web_ttt
-  plug :put_secret_key_base
   plug Plug.Logger
   plug :match
-  plug :create_or_find_game
   plug :dispatch
 
-  get ("/tictactoe/options") do
-    response_body = View.game_options()
+  get ("/ttt/options") do
+    response_body = View.stringified_game_options()
     conn |> put_resp_content_type("html") |> resp(200, response_body)
   end
 
-  get ("/tictactoe/play") do
-    game_state = conn |> GameSessionPlug.get_game_state()
-    response_body = View.render_game(game_state)
+  get ("ttt/play/:stringy_game_state") do
+    game_state = GameStateStringifier.parse(stringy_game_state)
+    response_body = View.render_game(game_state, stringy_game_state)
     conn |> put_resp_content_type("html") |> resp(200, response_body)
   end
 
-  get ("/tictactoe/computer_move") do
-    conn |> GameSessionPlug.make_next_move() |> redirect_to("/tictactoe/play")
+  get ("/ttt/computer_move/:stringy_game_state") do
+    old_game_state = GameStateStringifier.parse(stringy_game_state)
+    new_game_state = Game.make_next_move(old_game_state)
+    stringy_new_game_state = GameStateStringifier.stringify(new_game_state)
+    conn |> redirect_to("/ttt/play/#{stringy_new_game_state}")
   end
 
-  post ("/tictactoe/moves/:move") do
-    conn |> GameSessionPlug.update_game_state_with_move(move) |> redirect_to("/tictactoe/play")
+  post ("/ttt/moves/:move/:stringy_game_state") do
```

Comment:
mm interesting, in my head, the reason for using POST is  that the client is sending information to the server, which then updates the state and redirects to the main playing page (with the new state embedded)...though I guess I am going against the convention of using POST only for a  request where the information changes the state of the server? In which case should I change this route to a GET or is it a more fundamental problem?